pkg/authentication/handlers.go

```diff
@@ -53,26 +76,57 @@ type API struct {
 	apiKey           string
 	payloadValidator validation.PayloadValidatorInterface
 	oauth2           OAuth2ContextInterface
+	helper           OAuth2HelperInterface
+	authCookiesTTL   time.Duration
 
-	tracer   trace.Tracer
-	logger   logging.LoggerInterface
-	nonceTTL time.Duration
+	tracer        trace.Tracer
+	logger        logging.LoggerInterface
+	cookieManager AuthCookieManagerInterface
 }
 
 func (a *API) RegisterEndpoints(mux *chi.Mux) {
 	mux.Get("/api/v0/login", a.handleLogin)
+	mux.Get("/api/v0/auth/callback", a.handleCallback)
 }
 
 func (a *API) handleLogin(w http.ResponseWriter, r *http.Request) {
 	// add the Otel HTTP Client
 	r = r.WithContext(OtelHTTPClientContext(r.Context()))
 
-	code := r.URL.Query().Get("code")
+	nonce := a.helper.RandomURLString()
+	state := a.helper.RandomURLString()
+
+	a.cookieManager.SetNonceCookie(w, nonce, a.authCookiesTTL)
+	a.cookieManager.SetStateCookie(w, state, a.authCookiesTTL)
```

Comment:
question: does this belong on the handler layer? I think that authn request param generation should be done in another place

Reply:
The generation is in fact delegated to the helper object.
Setting cookies is 100% controller layer logic so it belongs in the handler, that's why the controller uses the helper.